### PR TITLE
Produce .csproj files with SplitPackages tool

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #134 

cc @natemcmaster @pranavkm @javiercn

### CODE REVIEW

File: src/SplitPackages/CsprojFileBuilder.cs [hunk 1]
```diff
@@ -114,39 +113,79 @@ public void AddImports(string fx, string import)
...
-            document["runtimes"] = JObject.FromObject(CreateRuntimesDictionary());
-
-            var writer = _whatIf ? StreamWriter.Null : File.CreateText(_path);
-            using (var jsonWriter = new JsonTextWriter(writer) { Formatting = Formatting.Indented })
+            var document = new XDocument();
+            var dependenciesDictionary = CreateDependenciesDictionary(_dependencies);
+            var packageDependencies = CreatePackageReferenceList(dependenciesDictionary);
+            var root = new XElement("Project", new XAttribute("Sdk", "Microsoft.NET.Sdk"), new XAttribute("ToolsVersion", "15.0"));
+            var propertyGroup = new XElement("PropertyGroup");
+            propertyGroup.Add(new XElement("TargetFrameworks", string.Join(";", _frameworks.Select(i => Frameworks.GetMoniker(i.Name)))));
```
Comment (javiercn) [inline src/SplitPackages/CsprojFileBuilder.cs]: Move string.Join(...) into its own variable.

File: src/SplitPackages/CsprojFileBuilder.cs [hunk 2]
```diff
@@ -114,39 +113,79 @@ public void AddImports(string fx, string import)
...
             {
-                _logger.LogInformation($"Writing project.json file to {_path}");
-                document.WriteTo(jsonWriter);
+                var monikerName = Frameworks.GetMoniker(framework.Name);
+                if (!monikerName.StartsWith("netcoreapp"))
+                {
+                    var runtimes = Runtime.AllRuntimes.ToList();
+                    if (runtimes != null && runtimes.Count != 0)
+                    {
+                        propertyGroup.Add(new XElement("RuntimeIdentifiers", new XAttribute("Condition", $" '$(TargetFramework)' == '{monikerName}'"), string.Join(";", runtimes.Select(i => i.Name))));
```
Comment (javiercn) [inline src/SplitPackages/CsprojFileBuilder.cs]: Break this line

File: src/SplitPackages/CsprojFileBuilder.cs [hunk 3]
```diff
@@ -114,39 +113,79 @@ public void AddImports(string fx, string import)
...
+                    }
+                }
+
+                var frameworkDependencies = CreateFrameworksDictionary(framework.Dependencies);
+                var imports = framework.Imports;
+
+                if (imports != null && imports.Count != 0)
+                {
+                    propertyGroup.Add(new XElement("PackageTargetFallback", new XAttribute("Condition", $" '$(TargetFramework)' == '{monikerName}'"), $"$(PackageTargetFallback);{string.Join(";", imports.Select(i => Frameworks.GetMoniker(i)))}"));
```
Comment (javiercn) [inline src/SplitPackages/CsprojFileBuilder.cs]: Break this line

File: src/SplitPackages/CsprojFileBuilder.cs [hunk 4]
```diff
@@ -114,39 +113,79 @@ public void AddImports(string fx, string import)
...
+
+            foreach (var framework in _frameworks)
             {
-                _logger.LogInformation($"Writing project.json file to {_path}");
-                document.WriteTo(jsonWriter);
+                var monikerName = Frameworks.GetMoniker(framework.Name);
+                if (!monikerName.StartsWith("netcoreapp"))
+                {
+                    var runtimes = Runtime.AllRuntimes.ToList();
+                    if (runtimes != null && runtimes.Count != 0)
```
Comment (javiercn) [inline src/SplitPackages/CsprojFileBuilder.cs]: Can't you do runtimes?.Count != 0?

File: src/SplitPackages/CsprojFileBuilder.cs [hunk 5]
```diff
@@ -114,39 +113,79 @@ public void AddImports(string fx, string import)
...
+                    {
+                        propertyGroup.Add(new XElement("RuntimeIdentifiers", new XAttribute("Condition", $" '$(TargetFramework)' == '{monikerName}'"), string.Join(";", runtimes.Select(i => i.Name))));
+                    }
+                }
+
+                var frameworkDependencies = CreateFrameworksDictionary(framework.Dependencies);
+                var imports = framework.Imports;
+
+                if (imports != null && imports.Count != 0)
```
Comment (javiercn) [inline src/SplitPackages/CsprojFileBuilder.cs]: ?.

File: src/SplitPackages/CsprojFileBuilder.cs (hunk 1 above)
Comment (javiercn) [inline src/SplitPackages/CsprojFileBuilder.cs]: Give this variable a meaningful name like projectDefinitionPropertyGroup or simply projectDefinition but something that indicates what you are trying to achieve with this. The problem is property group is very generic and makes it hard to know what its being used for.

File: src/SplitPackages/CsprojFileBuilder.cs [hunk 6]
```diff
@@ -114,39 +113,79 @@ public void AddImports(string fx, string import)
...
 
-                if (_whatIf)
+            root.Add(propertyGroup);
+            root.Add(new XElement("ItemGroup", packageDependencies));
+            foreach (var framework in _frameworks)
+            {
+                var itemGroup = new XElement("ItemGroup");
+                var frameworkDependencies = CreateFrameworksDictionary(framework.Dependencies);
+                var monikerName = Frameworks.GetMoniker(framework.Name);
+                if (frameworkDependencies != null && frameworkDependencies.Count != 0)
```
Comment (javiercn) [inline src/SplitPackages/CsprojFileBuilder.cs]: I would move the whole condition right before the item group and add it to the document only if there are dependencies specific to that framework.

**Comment by jbagga** [comment]
@javiercn Updated

File: src/SplitPackages/CsprojFileBuilder.cs [hunk 7]
```diff
@@ -114,39 +113,87 @@ public void AddImports(string fx, string import)
 
         public void Execute()
         {
-            var document = new JObject();
-            document["dependencies"] = JObject.FromObject(CreateDependenciesDictionary(_dependencies));
-            document["frameworks"] = JObject.FromObject(CreateFrameworksDictionary());
-            document["runtimes"] = JObject.FromObject(CreateRuntimesDictionary());
+            var document = new XDocument();
+            var root = new XElement("Project", new XAttribute("Sdk", "Microsoft.NET.Sdk"), new XAttribute("ToolsVersion", "15.0"));
```
Comment (natemcmaster) [inline src/SplitPackages/CsprojFileBuilder.cs]: Nit: we don't need to add the ToolsVersion attribute anymore (was a recent change.)

**Comment by natemcmaster** [comment]
Have you had a chance to look at this lately?

**Comment by jbagga** [comment]
I am waiting for @javiercn to check in his changes for other tools #169, #170 and #171. Once he does that and signs this off, I can check this in.

**Comment by natemcmaster** [comment]
Ok, good to know.

**Pull request merged**